cmd/snap-exec/main.go

```diff
@@ -92,36 +93,49 @@ func run() error {
 		return execHook(snapApp, revision, opts.Hook)
 	}
 
-	return execApp(snapApp, revision, opts.Command, extraArgs)
+	return execApp(snapApp, revision, opts.Command, extraArgs, opts.SkipCommandChain)
 }
 
 const defaultShell = "/bin/bash"
 
-func findCommand(app *snap.AppInfo, command string) (string, error) {
-	var cmd string
+func findCommand(app *snap.AppInfo, command string, skipCommandChain bool) (string, error) {
+	chain := make([]string, 0, len(app.CommandChain)+1)
+
+	if !skipCommandChain {
+		for _, element := range app.CommandChain {
+			chain = append(chain, filepath.Join(app.Snap.MountDir(), element))
```

Comment:
please save the mountdir in a variable instead of computing it n+1 times.

Also: this will be wrong as soon as we allow elements in the chain to refer to absolute paths in core (like `/bin/sh`), which is something we want to do.

Reply:
This functionality already exists today, this PR just moves it into this function. It will need to be changed once we allow absolute paths anyway.

Reply:
yes, sorry,  the second half was more a "try ot keep it contained or we'll be making our lives harder" thing

Reply:
Sure, done.